src/levanter/infra/ray_tpu.py

```diff
@@ -93,10 +101,87 @@ def do_run(remote_fn) -> _TpuRunResult:
                     logger.exception("Failed to kill job after primary failure")
             return _handle_ray_error(info, e)
 
-    return do_run.remote(remote_fn)
+    return do_run.remote(docker_cmd, name)
+
+
+def run_on_pod_multislice(docker_cmd: DockerRunCommand, name: str, tpu_type: str, num_slices: int) -> ray.ObjectRef:
```

Comment:
could we maybe just always make the dockerruncommand inject the MEGASCALE environment variables and use ray runtime environments to inject the environment variables that way?